shacl12-inf-rules/index.html

```diff
@@ -160,444 +223,1880 @@
+          </dt>
+          <dd>
+            A <em>rule body</em> is a sequence where each element
+            of the sequence is a [=triple pattern=], a [=condition expression=] or an [=assignment=].
+          </dd>
+          
+          <dt><dfn>rule set</dfn></dt>
+          <dd>
+            A <em>rule set</em> is a collection of zero or more rules.
+          </dd>
+        </dl>
+
+        <p>
+          In a [=triple pattern=] or a [=triple template=],
+          position 1 of the tuple is informally called the <em>subject</em>;
+          position 2 is informally called the <em>predicate</em>
+          position 3 is informally called the <em>object</em>.
+        </p>
+
+        <section id="wellformed">
+          <h3>Well-formedness Conditions</h3>
+          <p>
+            Well-formedness is a set of conditions on abstract syntax of
+            shapes rules.  Together, these rules ensure that a [=variable=] in a
+            [=head=] of a rule has a value defined in the [=body=] of the rule; 
+            that each variable in an condition expression or assignment 
+            expression has a value at the point of evaluation, and that each
+            assignment in a rule only introduces a new variable, 
+            not used earlier in the rule body.
+          </p>
+          <p>
+            A [=rule=] is a <dfn>well-formed rule</dfn> if all the following
+            conditions are met:
+          </p>
+          <ul>
+            <li>For every [=variable=] appearing in a [=triple templates=]
+              of the [=head=] of the [=rule=],
+              there is one or more occurrences of a [=variable=] 
+              of the same name in the [=triple patterns=] in the [=body=], 
+              or in an [=assignment=] in the body, or both.
+            </li>
+            <li>
+              For every [=variable=] in an [=expression=] at position <em>i</em>
+              of the [=body=], there is a corresponding [=variable=] of the same
+              name occuring in a [=triple pattern=], or occurring as an
+              [=assignment variable=], at a position <em>j</em>, where <em>j
+              &lt; i</em>.
+            </li>
+            <li>
+              An [=assignment variable=] is used in only one [=assignment=] in the [=body] of the [=rule=].
+            </li>
+            <li>
+              An [=assignment variable=] at position <em>i</em> of a [=rule body=] does not
+              occur in any [=triple pattern=] at position <em>j</em> where <em>j &lt; i</em>.
+            </li>
+            <li>
+              For every [=variable=] in an [=assignment expression=] at position <em>i</em>,
+              there is a corresponding variable in a triple pattern at position 
+              <em>j</em> where <em>j &lt i</em>,
+              or there is an [=assignment variable=] in an [=assignment=] at position 
+              <em>j</em> where <em>j &lt i</em>.
```

Comment:
Again, it seems that `j` _should be_ greater than `i`, here.
```suggestion
              For every [=variable=] in an [=assignment expression=] at position <em>i</em>,
              there is a corresponding variable in a triple pattern,
              or an [=assignment variable=] in an [=assignment=],
              at position 
              <em>j</em>, where <em>j</em> is less than <em>i</em>.
```

Reply:
Would you prefer i is greater than j?

Reply:
Currently changed to `i > j`.

Reply:
As above. I prefer the words to the symbol, but I can live with the symbol.